src/Stoa.ts

```diff
@@ -147,18 +156,97 @@ class Stoa {
                 return;
             }
 
-            console.log(block);
+            // Extract the block height from JSON.
+            let getBlockHeight = (block: any): number =>
+            {
+                if  (
+                    (block.header !== undefined) &&
+                    (block.header.height !== undefined) &&
+                    (block.header.height.value !== undefined)
+                )
+                {
+                    return Number(block.header.height.value);
+                }
+                else
+                {
+                    throw Error("Not found block height in JSON Block");
+                }
```

Comment:
A style note: It is better to get the error cases "out of the way". It also allows to remove nesting.
In this case, since `if` will `return`, you can just write:
```suggestion
                if  (
                    (block.header !== undefined) &&
                    (block.header.height !== undefined) &&
                    (block.header.height.value !== undefined)
                )
                {
                    return Number(block.header.height.value);
                }
                throw Error("Not found block height in JSON Block");
```

But then, it's a bit odd, hence my recommendation to handle errors first:
```suggestion
                if  (error_condition)
                    throw Error("Not found block height in JSON Block");
                return Number(block.header.height.value);
```